omod/src/main/webapp/resources/htmlforms/072-IntegratedMaternityRegister.xml

```diff
@@ -1,427 +1,661 @@
+	            } else {
+	                fieldHelper.enable($infantFeedingOptionField);
+	            }
+	        };
+
+	        var arvsToMotherChange =  function() {
+	            var $arvsToMother =  getValue('arvs-to-mother.value');
+	            var $preArtNumberField = jq(PRE_ART_NUMBER_FIELD).find('input:text');
+	            if ($arvsToMother != '') {
+	                fieldHelper.enable($preArtNumberField);
+	            } else {
+	                fieldHelper.disable($preArtNumberField);
+	            }
+	        };
+
+	        var $othersField = jq('span.has-comment input[type=checkbox]');
+
+	        var toggleOtherSpecify = function() {
+	            $othersField.each(function(){
+	                var $specifyField = jq(this).siblings('input[type=text]');
+	                if (jq(this).attr('checked') == 'checked') {
+	                    fieldHelper.removeReadonly($specifyField);
+	                } else {
+	                    fieldHelper.makeReadonly($specifyField);
+	                    $specifyField.val('');
+	                }
+	            });
+	        };
+
+	        var viralLoadResultsValidate = function() {
+	            var $viralLoadResultsQualitative = getValue('viral-load-results.value');
+	            var $viralLoadResultsQuantitative = getValue('viral-load.value');
+
+	            if($viralLoadResultsQualitative == VIRAL_LOAD_DETECTED_CONCEPT_ID &amp; $viralLoadResultsQuantitative == '') {
+	                getField('viral-load.error').html('Invalid viral load results').show();
+	                getField('viral-load.value').addClass('error');
+	                return false;
+	            } else {
+	            	getField('viral-load.value').removeClass('error');
+	            }
+
+	            return true;
+	        }
+
+		    var isOtherSelectedAndSpecified = function() {
+		    	var valid = true;
+		        $othersField.each(function(){
+		            var $specifyField = jq(this).siblings('input[type=text]');
+		            var $errorField = jq(this).siblings('span.error');
+		            if (jq(this).attr('checked') == 'checked' &amp;  $specifyField.val() == '') {
+		                $errorField.html('Please specify').show();
+		                $specifyField.addClass('error');
+		                valid = false;
+		            } else {
+		            	$errorField.html('').hide();
+		            	$specifyField.removeClass('error');
+		            }
+
+		        });
+
+		        return valid;
+		    };
+
+	        $othersField.change(toggleOtherSpecify);
+
+	        $viralLoadResultsField.change(viralLoadResultsChange);
+	        $hivStatusField.change(hivStatusChange);
+	        $iycfField.change(iycfChange);
+	        $arvsToMotherField.change(arvsToMotherChange);
+
+	        viralLoadResultsChange();
+	        hivStatusChange();
+	        iycfChange();
+	        arvsToMotherChange();
+	        toggleOtherSpecify();
+
+	        beforeSubmit.push(function(){

```

Comment:
@ssmusoke The validation message  are set within the functions that perform validation. Case in point:

	        ```
var viralLoadResultsValidate = function() {
	            var $viralLoadResultsQualitative = getValue('viral-load-results.value');
	            var $viralLoadResultsQuantitative = getValue('viral-load.value');

	            if($viralLoadResultsQualitative == VIRAL_LOAD_DETECTED_CONCEPT_ID &amp; $viralLoadResultsQuantitative == '') {
	                **_getField('viral-load.error').html('Invalid viral load results').show();
	                getField('viral-load.value').addClass('error');_**
	                return false;
	            } else {
	            	**_getField('viral-load.value').removeClass('error');_**
	            }

	            return true;
	        }
```